File: java/socketd-transport/socketd-transport-smartsocket/src/main/java/org/noear/socketd/transport/smartsocket/TcpAioClient.java
package org.noear.socketd.transport.smartsocket;

import org.noear.socketd.transport.client.ClientBase;
import org.noear.socketd.transport.client.ClientConfig;
import org.noear.socketd.transport.client.ClientConnector;
import org.noear.socketd.transport.core.ChannelSupporter;
import org.noear.socketd.transport.smartsocket.impl.FrameProtocol;
import org.smartboot.socket.transport.AioSession;

/**
 * Tcp-Aio 客户端实现
 *
 * @author noear
 * @since 2.0
 */
public class TcpAioClient extends ClientBase<TcpAioChannelAssistant> implements ChannelSupporter<AioSession> {
    private final FrameProtocol frameProtocol;
    public FrameProtocol frameProtocol() {
        return frameProtocol;
    }

    public TcpAioClient(ClientConfig config) {
        super(config, new TcpAioChannelAssistant(config));
        this.frameProtocol = new FrameProtocol(this);
    }

    @Override
    protected ClientConnector createConnector() {
        return new TcpAioClientConnector(this);
    }
}

File: python/socketd/socketd_aio_tcp/TcpAioClient.py
import asyncio

from socketd.transport.core.ChannelSupporter import ChannelSupporter
from socketd.transport.client.ClientBase import ClientBase
from socketd.transport.client.ClientConfig import ClientConfig

from .TcpAioClientConnector import TcpAioClientConnector
from .TcpAIOChannelAssistant import TcpAIOChannelAssistant


class TcpAioClient(ClientBase, ChannelSupporter):

    def __init__(self, client_config: ClientConfig):
        ClientBase.__init__(self, client_config, TcpAIOChannelAssistant(client_config, asyncio.get_event_loop()))

    def create_connector(self) -> TcpAioClientConnector:
        return TcpAioClientConnector(self)
